extensions/s3/src/main/java/io/deephaven/extensions/s3/S3InstructionsProviderPlugin.java

```diff
@@ -0,0 +1,52 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.extensions.s3;
+
+import com.google.auto.service.AutoService;
+import io.deephaven.util.channel.DataInstructionsProviderPlugin;
+import org.jetbrains.annotations.NotNull;
+
+import java.net.URI;
+import java.util.Map;
+
+/**
+ * {@link DataInstructionsProviderPlugin} implementation used for reading files from S3.
+ */
+@AutoService(DataInstructionsProviderPlugin.class)
+@SuppressWarnings("unused")
+public final class S3InstructionsProviderPlugin implements DataInstructionsProviderPlugin {
+    private static final String CLIENT_REGION = "client.region";
+    private static final String S3_ACCESS_KEY_ID = "s3.access-key-id";
+    private static final String S3_SECRET_ACCESS_KEY = "s3.secret-access-key";
+    private static final String S3_ENDPOINT = "s3.endpoint";
+
+    @Override
+    public Object createInstructions(@NotNull URI uri, @NotNull Map<String, String> properties) {
+        // If the URI scheme is "s3" or "s3a" or if the properties contain one of these specific keys, we can create a
+        // useful S3Instructions object.
+        if (uri.getScheme().equals("s3")
```

Comment:
I think we might want to think about going the opposite direction; if we have an S3Instructions, we can populate properties. We can compare, contrast with https://github.com/deephaven/deephaven-core/pull/5880

Reply:
Most other Iceberg implementations (pyiceberg, Java API) use the property collection method. I think transitioning users will be more accustomed to this method (although we don't use this anywhere else in DH, so we can choose to be internally consistent).

Since the goal is to prevent duplication of these properties in the Catalog and the S3Instructions, either method is acceptable IMO.